src/main/webapp/app/exercises/programming/shared/lifecycle/programming-exercise-lifecycle.component.html

```diff
@@ -1,7 +1,7 @@
 <div class="timeline-box">
     <div class="test-schedule">
         <div class="test-schedule-row d-flex justify-content-around">
-            @if (!isExamMode) {
+            @if ((isEditFieldDisplayedRecord()?.releaseDate || !isEditFieldDisplayedRecord()) && !isExamMode) {
```

Comment:
Did you think about restructuring the if statements? 
Since you're checking multiple times for ... `|| !isEditFieldDisplayedRecord()) && !isExamMode`

Reply:
@krusche specified that the configuration of the `simple mode` should be easily adjustable. This is why checks are not combined, even if possible, as it allows to configure the available configuration options easily via a simple datastructure `export type InputFieldEditModeMapping = Record<ProgrammingExerciseInputField, boolean>;`

Reply:
I did not clarify this in the PR description